src/lsptoolshost/razor/showGeneratedDocumentCommand.ts

```diff
@@ -0,0 +1,91 @@
+﻿/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { RazorLanguage } from '../../razor/src/razorLanguage';
+import { getUriPath } from '../../razor/src/uriPaths';
+import { RoslynLanguageServer } from '../server/roslynLanguageServer';
+import { DocumentContentsRequest } from './documentContentsRequest';
+import { CancellationToken, RequestType, TextDocumentIdentifier } from 'vscode-languageserver-protocol';
+import { UriConverter } from '../utils/uriConverter';
+import { GeneratedDocumentKind } from './generatedDocumentKind';
+
+export class ShowGeneratedDocumentCommand {
+    private static requestType = new RequestType<DocumentContentsRequest, string, void>(
+        'razor/generatedDocumentContents'
+    );
+
+    public static register(roslynLanguageServer: RoslynLanguageServer) {
+        return vscode.Disposable.from(
+            vscode.commands.registerCommand('extension.showRazorCSharpWindow', async () =>
+                this.show(roslynLanguageServer, GeneratedDocumentKind.CSharp, '.g.cs')
+            ),
+            vscode.commands.registerCommand('extension.showRazorHtmlWindow', async () =>
+                this.show(roslynLanguageServer, GeneratedDocumentKind.Html, '.g.html')
+            )
+        );
+    }
+
+    private static async show(
+        roslynLanguageServer: RoslynLanguageServer,
+        kind: GeneratedDocumentKind,
+        extension: string
+    ) {
+        const uri = await this.getActiveDocumentUri();
```

Comment:
it might not work here, but the spec added a document content request - https://microsoft.github.io/language-server-protocol/specifications/lsp/3.18/specification/#workspace_textDocumentContent

This is done based on URI - and when given such a URI, the client will automatically request the content from the server and display a readonly version of the content.  

You seem to be doing more here with webviews, and you also need to map the Razor URI to the generated document, but just a thought.

Reply:
Yeah, I saw the related Roslyn issue and have been watching it, but at the end of the day it's just a bit too clunky for our needs. The client doesn't know the C# file name, and ironically the server doesn't know the Html file name. We could do a `workspace/textDocumentContent` request for some constructed fake C# uri, or we could exchange more info backwards and forwards to each party knew about the right uris, but I just don't think there is a big enough benefit. The web view is a bit crap, but its all these commands ever did, and they're not commands that people use regularly. I was going to leave them out altogether with cohosting, but someone on the C# Discord asked about the VS syntax visualizer, and making that work made this new endpoint pretty trivial to implement.